### ISSUE TITLE
Crash (SIGABRT) when fast scrolling or slow network in UITableView

### ISSUE THREAD

**Issue by robinmonjo** [rs/SDWebImage#1375]
I use SDWebImageView to load some images inside a UITableViewCell subclass. It all works pretty well except when I scroll fast or when the network is slow.

It looks like when the cell get recycled or dealloc, the loading of the image continue and crash at some points. Here is some more information

### UITableViewCell subclass

`````objective-c
- (void) configureCellWithObjt:(SomeObj *)obj {
  self.titleLabel.text = obj.title;
  self.subtitleOneLabel.text = obj.subtitle;
    [self.avatarImageView sd_setImageWithURL:[NSURL URLWithString:obj.avatarURLMedium] placeholderImage:[UIImage imageNamed:@"anon_face_thumb"]];
}
`````

This method is called in the tableView's `cellForRowAtIndexPath`. When I scroll fast, I get this crash:

![screen shot 2015-11-27 at 10 43 18](https://cloud.githubusercontent.com/assets/1815076/11438249/e74cdab8-94f3-11e5-92a8-a74d63a7eaff.png)

This happens in the `SDWebImageViewDownloaderOperation.m` line 96 (as of version 3.7.3).

I tried to cancel the operation (`sd_cancelCurrentImageLoad`) when the cell get recycled or hidden, but it doesn't fix my crash.

Any ideas, on something I might do wrong ? or maybe it's a bug ?

Regards

**Comment by mythodeia** [comment]
does this occur in the demo project as well?

**Comment by AnkitGupta87** [comment]
Have been facing this issue in our App Store build at a very large scale.. unable to reproduce on our end

**Comment by robinmonjo** [comment]
To reproduce this, on a table view with 230 cell, I just scroll really fast in both directions (down and top) and this eventually happens. My images are loaded from a UITableViewCell subclass, will check if there is a similar demo project.

**Comment by xinwo** [comment]
same  proplem here. 
 any update?

**Comment by AnkitGupta87** [comment]
any updates?

**Comment by robinmonjo** [comment]
There is no samples with SDWebImageView getting called from within a custom UITableViewCell, so for me it's a bug from the library. Won't have time to dig more though.

**Comment by robinmonjo** [comment]
Limiting the number of concurrent fetches with `SDWebImageManager.sharedManager.imageDownloader.maxConcurrentDownloads = 2;` solved the issue for me.

**Comment by bpoplauschi** [comment]
Let's close this issue, see if it happens again with 4.0.0 (4.0.0-beta is available for download, 4.0.0 official release is on its way).
